applications/minotari_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -311,6 +311,27 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
                     let randomx_estimated_hash_rate = randomx_hash_rate_moving_average.average();
                     let estimated_hash_rate = sha3x_estimated_hash_rate.saturating_add(randomx_estimated_hash_rate);
 
+                    let block = match handler.get_block(current_height, true).await {
+                        Ok(block) => block,
+                        Err(err) => {
+                            warn!(target: LOG_TARGET, "Base node service error: {:?}", err,);
+                            let _network_difficulty_response = tx.send(Err(obscure_error_if_true(
+                                report_error_flag,
+                                Status::internal(format!("Error fetching block at height {}", current_height)),
+                            )));
+                            return;
+                        },
+                    };
+                    if block.is_none() {
+                        let _network_difficulty_response = tx.send(Err(obscure_error_if_true(
+                            report_error_flag,
+                            Status::internal(format!("Block not found at height {}", current_height)),
+                        )));
+                        return;
+                    }
+                    let block = block.unwrap();
+                    let coinbases = block.block().body.get_coinbase_outputs();
```

Comment:
If we only need the coinbase extra(s), the function could only return that?

Reply:
I thought it might be useful in future